test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -183,24 +187,51 @@ func schedulePods(config *testConfig) int32 {
 	}
 }
 
-// mutateNodeTemplate returns the modified node needed for creation of nodes.
-func (na nodeAffinity) mutateNodeTemplate(node *v1.Node) {
+// mutateTemplate returns the modified node and pod needed for creation of nodes.
+func (c schedulerPerfConfig) mutateTemplate(node *v1.Node, pod *v1.Pod, config *testConfig) {
+	// If use default provider, process predicates and priorities and return
+	if isDefaultProvider() {
+		processMatchNodeSelector(node, pod)
+		return
+	}
+
+	// Otherwise, process predicates and priorities based on configure file.
+	configFile, _ := ioutil.ReadFile(policyConfigFile)
+	var data map[string]interface{}
+	err := json.Unmarshal(configFile, &data)
+	if err != nil {
+		glog.Fatalf("failed to unmarshal configure file: %v", err)
+	}
+
+	predicateList := data["predicates"].([]interface{})
+
+	for _, predicate := range predicateList {
+		predicateName := predicate.(map[string]interface{})["name"]
+		// TODO(resouer): deal with more predicates & priorities
+		switch predicateName.(string) {
```

Comment:
We cannot have a switch here because, say if we want a node to have 2 different priorities/ predicates, this won't work. For example, if we want a node to be tainted/labelled for node Affinity, this may not work.

Reply:
`predicateList` contains all predicates enabled in this test, and the logic here is 

```
switch predicateName { 
  case: predicate1
    deal with predicate1
  case: predicate2
    deal with predicate2
}
```
Sorry, did not get what's wrong here.

Reply:
Apologies, I was in haste when typing comments. This would work.

Reply:
Thanks, I will make `labelCount` etc work and re-push.

Reply:
@resouer Yes. FWIW, the more configurable for end-user, the easier it would be to use scheduler_perf, for example in case of taints, if we can specify to use specific taint like requiredDuringScheduling etc, it would be easier to use. So, in my opinion structs for individual preds/priorities is worth having, unless you have something better in your mind.

@jayunit100 WDYT?

Reply:
Yea, I agree here.

Reply:
Have to hold this up for a few days, working on 1.9 feature recently